### ISSUE TITLE
Does not compile for Visual Studio 2015

### ISSUE THREAD

**Issue by kentbarber** [aws/aws-sdk-cpp#1313]
### What platform/OS are you using?
Windows 10

### Which version of the SDK?
1.7.266

### What compiler are you using? what version?
Visual Studio 14  2015 Win64  against Windows SDK 8.1

### What are your CMake arguments?
`cmake ../ -G "Visual Studio 14 2015 Win64" -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DBUILD_ONLY="s3;transfer" -DENABLE_TESTING=OFF -DENABLE_RTTI=OFF -DCMAKE_SYSTEM_VERSION=8.1 -Tv140`

### Can you provide a TRACE level log? (sanitize any sensitive information)
I can successfully build the aws-sdk-cpp for VS 2017 using "Visual Studio 15 2017 Win64". But when I try to build it for 2015 it has problems with the .deps. So I have been trying to create the project files for aws-c-common manually and run into the same problem. It can't find the compiler "rc.exe". 

Has anyone else been able to create the project files for VS 2015 recently? That is when you have VS 2017 (and 2019 in my case) also installed on your system.

I have a lot of projects that I compile via Jenkins on every VS platform, using CMake, and none of them have any issues at all with using -G "Visual Studio 14 2015 Win64". So I am just wondering if something is incorrect in these dependency projects.

This is what I am calling from within a build folder in aws-sdk-cpp

`cmake ../ -G "Visual Studio 14 2015 Win64" -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DBUILD_ONLY="s3;transfer" -DENABLE_TESTING=OFF -DENABLE_RTTI=OFF -DCMAKE_SYSTEM_VERSION=8.1 -Tv140`

To me it looks like the dependencies are being compiled from CMake? Is there a way to get it to make the Visual Studio project files for the dependencies that I can then compile myself?

----

With the error message being the following...

```
-- TARGET_ARCH not specified; inferring host OS to be platform compilation target
-- Building AWS libraries as shared objects
-- Generating windows build config
-- Building project version: 1.7.266
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:7 (project):
  No CMAKE_C_COMPILER could be found.

CMake Error at CMakeLists.txt:7 (project):
  No CMAKE_CXX_COMPILER could be found.

-- Configuring incomplete, errors occurred!
See also "F:/aws-sdk-cpp/build/.deps/CMakeFiles/CMakeOutput.log".
See also "F:/aws-sdk-cpp/build/.deps/CMakeFiles/CMakeError.log".
CMake Error at CMakeLists.txt:229 (message):
  Failed to configure third-party libraries.

```
-----------------------

And the error in the log file being

`  LINK : fatal error LNK1158: cannot run 'rc.exe' [F:\aws-sdk-cpp\build\.deps\CMakeFiles\3.11.2\CompilerIdCXX\CompilerIdCXX.vcxproj]`

----

I have also, as mentioned,  tried compiling the aws-c-common dependency directly. And also did so from a VS2015 x64 Native Tools command prompt after also running the vcvarsall.bat command as follows 

- "c:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" amd64 8.1
- cmake -G "Visual Studio 14 2015 Win64" ../aws-c-common

I then get the same error as above.

`LINK : fatal error LNK1158: cannot run 'rc.exe' [F:\aws-c-common-build\CMakeFiles\3.11.2\CompilerIdC\CompilerIdC.vcxproj]`

But if I simply type "rc.exe" in the command prompt it finds the correct compiler. But CMake does not.

----

May be an error on my part, but interested to know from anyone else if they have been able to compile the latest version under VS 2015.

**Comment by KaibaLopez** [comment]
Hi @kentbarber,
I think you have a typo on your cmake arguments,
try `-G "Visual Studio 14 2015" -A x64` instead of `-G "Visual Studio 14 2015 Win64"`
That would at least explain the cmake error:
`-- TARGET_ARCH not specified; inferring host OS to be platform compilation target`

As for the dependencies, yea we are building them by default but you can turn that off by adding the cmake argument: `-DBUILD_DEPS=OFF`, keep in mind you'd still have to get them and build them manually, and then point cmake to their installed path.

Let me know if the typo was the problem.

**Comment by kentbarber** [comment]
Thanks for your feedback @KaibaLopez. I have always used `-G "Visual Studio 14 2015 Win64"`, and that is what my copy of CMake specifies to use as its generator input as well. 

Its interesting that I only get this `-- TARGET_ARCH` message with this particular project and no other ones.

I tried the argument you suggested but still the same problem. And the problem is definitely with the dependencies. So I can't even build them manually because of this exact same problem. I posted an issue in the aws-c-common repo as well. 

I just thought perhaps someone where might have hit this same issue and know of a solution.

**Comment by KaibaLopez** [comment]
Hi @kentbarber ,
Sorry I should have noticed this earlier but yea the sdk requires vs2015 minimum, so it wouldn't work for you if you need it to compile on a older version.

From our ReadMe:
> Minimum Requirements:
    Visual Studio 2015 or later

I'm glad this is not a blocking issue for you, and I'll be closing as this is not something we plan on changing, sorry it doesn't work for you.

**Comment by kentbarber** [comment]
Hi @KaibaLopez 

This is vs2015.

-G "Visual Studio 14 2015 Win64" 

That is for Visual Studio 2015. 

It does not compile for vs2015. Which as you mentioned is the minimum requirement. 

It only seems to compile for vs2017 from my findings.

**Comment by kentbarber** [comment]
This issue should not be closed since it is still a problem. Unless vs2015 is no longer supported?

**Comment by KaibaLopez** [comment]
my bad, misread.

**Comment by KaibaLopez** [comment]
ok, @kentbarber  sorry for the mistake on my part, 
I was finally able to reproduce the error and the problem seems to be that you had not properly installed the C++ compiler in Visual studio 2015, hence the errors:
```
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:7 (project):
  No CMAKE_C_COMPILER could be found.

CMake Error at CMakeLists.txt:7 (project):
  No CMAKE_CXX_COMPILER could be found.
```
Basically all I did to fix it was open Visual studio 2015 and add the C++ Tools for VS, which included Windows SDK 8.1 (which looks like you already have) and Windows Platform CRT. 

Let me know if this fixes the issue for you too, or if the cause was different for you.

**Comment by kentbarber** [comment]
Hi @KaibaLopez, thanks for checking and confirming that it compiles for you. 

Do you have any other versions of Visual Studio installed at the time? I think this may be the problem. And it is a known issue for some projects. 

I have VS 2013, 2015, 2017 and 2019 installed. All with C++ tool installed.

These work perfectly fine by using the -G switch in CMake for all of my other projects. However for aws-sdk-cpp it doesn't work. Which makes me think the CMake setup is perhaps just looking in some default location for some installed version of the compiler instead of using the passed in -G parameter.

I have everything installed for Visual Studio 2015, including the C++ Tools and Windows SDK 8.1. 

I may have to look into seeing if I can manually create my own project files for the dependencies by including the source files myself, and simply not use CMake at all for VS 2015.

**Comment by KaibaLopez** [comment]
@kentbarber yea the multiple VS versions could be a problem. Right now I only have 2019 and 2015 installed, but I'll try installing the rest too and see if I that causes an error.

Also, could you show me what your cmake shows for successful builds, preferably 2019, 2013 and without -DBUILD_DEPS=OFF? 
2019:
```
-- The C compiler identification is MSVC 19.23.28105.4
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.23.28105/bin/Hostx64/x64/cl.exe -- works
```
2015:
```
-- The C compiler identification is MSVC 19.0.24215.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
```

**Comment by kentbarber** [comment]
Thanks @KaibaLopez. I found the problem. 

I noticed that my version of CMake didn't have -G flag for 2019 so updated to the latest version of CMake. And then I tested it with 2015 also, and it now works.

Thanks again for taking the time to look into this.

I would add a recommendation to the readme to state that they should have the latest version of CMake. Mine must have been a couple of years old I think.
